index.js

```diff
@@ -145,16 +145,34 @@ function $asInteger (i) {
 }
 
 function $asNumber (i) {
+  if (i === 0) return '0'
+  if (!i || i === true) {
+    throw new Error('Cannot coerce to number')
+  }
+  if (i.constructor === Number) {
```

Comment:
can you do a `typeof` and save the value somewhere?

Reply:
```js
const serializer = build({
  type: 'integer'
})
serializer(new Number(55)) // 55
```

This is the only case in which an instance of Object *is* a number.

The most correct form is the `instanceof` check but it could be expensive